src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowCreate.cs

```diff
@@ -227,16 +172,125 @@ public override EditUpdateCellResult SetCell(int columnId, string newValue)
+            List<string> inColumns = new List<string>();
+            List<string> outColumns = new List<string>();
+            List<SqlParameter> sqlParameters = new List<SqlParameter>(); 
+            for (int i = 0; i < AssociatedObjectMetadata.Columns.Length; i++)
+            {
+                DbColumnWrapper column = AssociatedResultSet.Columns[i];
+                CellUpdate cell = newCells[i];
+                
+                // Add an out column if we're doing this for a command
+                if (forCommand)
+                {
+                    outColumns.Add($"inserted.{SqlScriptFormatter.FormatIdentifier(column.ColumnName)}");
+                }
+                
+                // Skip columns that cannot be updated
+                if (!column.IsUpdatable)
+                {
+                    continue;
+                }
+                
+                // Make sure a value was provided for the cell 
+                if (cell == null)
+                {
+                    // If there isn't a default, then fail 
+                    if (DefaultValues[i] == null)
+                    {
+                        throw new InvalidOperationException(SR.EditDataCreateScriptMissingValue);
+                    }
+                    
+                    // There is a default value, so trust the db will apply it
+                    continue;
+                }
+
+                // Add the input values
+                if (forCommand)
+                {
+                    // Since this script is for command use, add parameter for the input value to the list
+                    string paramName = $"@Value{RowId}{i}";
+                    inValues.Add(paramName);
+
+                    SqlParameter param = new SqlParameter(paramName, cell.Column.SqlDbType) {Value = cell.Value};
+                    sqlParameters.Add(param);
+                }
+                else
+                {
+                    // This script isn't for command use, add the value, formatted for insertion
+                    inValues.Add(SqlScriptFormatter.FormatValue(cell.Value, column));
+                }
+                
+                // Add the column to the in columns
+                inColumns.Add(SqlScriptFormatter.FormatIdentifier(column.ColumnName));
+            }
+            
+            // Begin the script (ie, INSERT INTO blah)
+            StringBuilder queryBuilder = new StringBuilder();
+            queryBuilder.AppendFormat(InsertScriptStart, AssociatedObjectMetadata.EscapedMultipartName);
+            
+            // Add the input columns (if there are any)
+            if (inColumns.Count > 0)
+            {
+                string joinedInColumns = string.Join(", ", inColumns);
+                queryBuilder.AppendFormat(InsertScriptColumns, joinedInColumns);
+            }
+            
+            // Add the output columns (this will be empty if we are not building for command)
+            if (outColumns.Count > 0)
```

Comment:
Minor: this is repeated logic. A `joinAndAppend` method could check count > 0 and append comma-delimited values to the builder.

Reply:
Since this fix still doesn't address triggers, I'm in the process of working on another change that will fix some of the repeated logic. As such, I'll leave this as-is for now.